src/main/java/com/conveyal/osmlib/OSM.java

```diff
@@ -75,23 +78,41 @@ public OSM (String diskPath) {
                 // 'direct' means off-heap memory, no garbage collection overhead
                 dbMaker = DBMaker.newMemoryDirectDB(); 
             } else {
-                LOG.info("OSM will be stored in file {}.", diskPath);
-                dbMaker = DBMaker.newFileDB(new File(diskPath));
+                File dp = new File(diskPath);
+                reading = diskPath.endsWith("mapdb") && dp.exists();
```

Comment:
Good point. Let's change it to `(diskPath.endsWith(".mapdb") || diskPath.endsWith(".db")`